tuxemon/event/conditions/common.py

```diff
@@ -0,0 +1,49 @@
+# SPDX-License-Identifier: GPL-3.0
+# Copyright (c) 2014-2023 William Edwards <shadowapex@gmail.com>, Benjamin Bean <superman2k5@gmail.com>
+import logging
+
+from tuxemon.npc import NPC
+
+logger = logging.getLogger(__name__)
+
+
+class CommonCondition:
+    name = "Common"
+
+    @staticmethod
+    def check_character_attribute(
+        character: NPC,
+        attribute: str,
+        value: str,
+    ) -> bool:
+        """
+        Check a character's (npc or player) attribute.
+
+        Parameters:
+            character: The NPC object to check.
+            attribute: The attribute to check.
+            value: Value to check.
+
+        """
+
+        # check for valid inputs
+        # trigger an AttributeError if the attribute doesn't already exist
+        attr = None
+        try:
+            attr = getattr(character, attribute)
+        except AttributeError:
+            logger.warning(
+                "Character attribute '{0}' specified does not exist.",
+                attribute,
+            )
+            return False
+
+        try:
+            val = type(attr)(value)
+        except TypeError:
+            logger.warning(
+                "The value given cannot be parsed into the correct type for '{0}'",
+                attribute,
+            )
+            return False
+        return bool(attr == val)
```

Comment:
Bool convertion is unnecessary there.

`==` operation is always going to return a `bool`.